src/main/java/seedu/mtracker/console/AddInstrumentParser.java

```diff
@@ -89,18 +91,41 @@ public static boolean isValidPrice(String currentPrice) {
         return isValid;
     }
 
-    public static boolean isExpiryFilled(String expiryInput) {
-        boolean isFilled = true;
+    public static void checkDateFormat(String expiryInput) throws InvalidDateError {
+        try {
+            LocalDate.parse(expiryInput);
+        } catch (DateTimeParseException e) {
+            throw new InvalidDateError();
+        }
+    }
+
+    public static void checkDateInPast(String expiryInput) throws InvalidDateError {
+        LocalDate givenDate = LocalDate.parse(expiryInput);
+        if (givenDate.equals(LocalDate.now()) || givenDate.isAfter(LocalDate.now())) {
+            return;
+        }
+
+        throw new InvalidDateError();

```

Comment:
`if (!givenDate.equals(LocalDate.now()) & !givenDate.isAfter(LocalDate.now())) {
throw new InvalidDateError();
}`

Reply:
alright!, btw no need to merge this first, can wait for the other to add their dg